src/shared-components/button/components/linkButton/index.tsx

```diff
@@ -66,8 +66,10 @@ export const LinkButton = ({
       // eslint-disable-next-line react/jsx-props-no-spreading
       {...rest}
     >
-      <ButtonContents hasIcon={false}>
-        <ButtonText>{children}</ButtonText>
+      <ButtonContents isLoading={false} isFullWidth={false} hasIcon={false}>
```

Comment:
Rather than rely on casting undefined props in `style.ts`, we explicitly pass in what is currently the default behavior for each component. This makes it clearer from the markup outside of `style.ts` what kinds of props the component expresses. 

We could also wrap `style.ts` props in `isDefined`, or use ES6 defaults in that file. I originally started doing that and didn't think it was very elegant, instead preferring a very explicit API. **It does not change the API for the consumer**, which was more important than breaking the internal API. But I'm open to feedback here.